### ISSUE TITLE
101-vm-secure-password does not reference keyvault

### ISSUE THREAD

**Issue by larryclaman** [Azure/azure-quickstart-templates#7757]
Regarding azure-quickstart-templates/tree/master/101-vm-secure-password, this template does not reference a key vault.

[Template Name goes here](Template link goes here)

### Issue Details
This template is described as "This template allows you to deploy a simple Windows VM by retrieving the password that is stored in a Key Vault. Therefore the password is never put in plain text in the template parameter file."

However, neither the current version of the template nor the parameters file references a keyvault.   It appears that the keyvault was (inadvertently) removed at commit #ffe854534869c876df07d60b78fcd70fc02f877a (pr #7230).  

As it stands, the template appears to deploy a standard VM and does not demonstrate the template use case.

**Comment by JohnathonMohr** [comment]
The modification was made to allow this template to pass CI/CD checks by conforming to the guidance in the [Contribution Guide](https://github.com/Azure/azure-quickstart-templates/tree/master/1-CONTRIBUTION-GUIDE#parameters-file-placeholders) (under Key Vault Related placeholders).

However, I do see this makes the demonstration of the template more difficult to understand/follow and replicate.

@bmoore-msft - what do you suggest for addressing this?

**Comment by larryclaman** [comment]
@JohnathonMohr seems like there's a disconnect between the placeholders that are specified in the Contribution Guide and the actual template.  When I try to deploy the template, I end up with a VM with a plain old password that does not seem to reference any keyvault during its configuration.  So it seems like the point of the ARM template sample is lost, in that it does not provide a reference to users as to how to author an arm template that retrieves a password from a Key Vault.

**Comment by JohnathonMohr** [comment]
@larryclaman yes, the template currently does not provide a valuable demonstration of how to retrieve the password from a KeyVault.  The template originally referenced the KeyVault in the parameters file (not the template itself), and the retrieved password was then passed to the template as a parameter.  Since the parameters file was changed to conform to the contribution guide, that KeyVault reference was lost.

Not saying this shouldn't be fixed, just want to clarify that the example demonstrated how to retrieve a KeyVault secret via the parameters, not within the template itself.

**Comment by JohnathonMohr** [comment]
@larryclaman I was able to fix the parameters file in the sample to bring back the intended demonstration.  Pull request is #7758.  Once @bmoore-msft signs off I can merge it.

**Comment by MCKLMT** [comment]
The pull request has been merged. The issue can be closed.
